elrond-wasm/src/storage/mappers/linked_list_mapper.rs

```diff
@@ -423,39 +487,48 @@ where
 impl<'a, SA, T> Iter<'a, SA, T>
 where
     SA: StorageReadApi + StorageWriteApi + ManagedTypeApi + ErrorApi + Clone + 'static,
-    T: TopEncode + TopDecode + 'static,
+    T: TopEncode + TopDecode + NestedEncode + NestedDecode + Clone + 'static,
 {
     fn new(linked_list: &'a LinkedListMapper<SA, T>) -> Iter<'a, SA, T> {
         Iter {
             node_id: linked_list.get_info().front,
             linked_list,
         }
     }
+
+    fn new_from_node_id(linked_list: &'a LinkedListMapper<SA, T>, node_id: u32) -> Iter<'a, SA, T> {
+        Iter {
+            node_id,
+            linked_list,
+        }
+    }
 }
 
 impl<'a, SA, T> Iterator for Iter<'a, SA, T>
 where
     SA: StorageReadApi + StorageWriteApi + ManagedTypeApi + ErrorApi + Clone + 'static,
-    T: TopEncode + TopDecode + 'static,
+    T: TopEncode + TopDecode + NestedEncode + NestedDecode + Clone + 'static,
 {
     type Item = T;
 
     #[inline]
     fn next(&mut self) -> Option<T> {
         let current_node_id = self.node_id;
+
         if current_node_id == NULL_ENTRY {
             return None;
         }
-        self.node_id = self.linked_list.get_node(current_node_id).next;
-        Some(self.linked_list.get_value(current_node_id))
+
+        let node = self.linked_list.get_node(current_node_id);
+        self.node_id = node.next_id;
+        Some(node.value)
     }
```

Comment:
This method should probably return the whole node now, otherwise it would be difficult to save the current node_id, for instance. (unless I'm missing something)